### PR TITLE
feat: add support for opdk

### PR DESCRIPTION
**Pull request by SamPriyadarshi**
## Description
- added new flag for opdk
-  basic authentication being used for opdk
-  added support for list, export and report operations for OPDK

## Housekeeping
(please check all that apply [x], do not edit the text)
- [x] I have run all the tests locally and they all pass.
- [x] I have followed the relevant style guide for my changes.
- [ ] PR requires full pipeline run (Run for changes only by default).

**CC:** @apigee-devrel-reviewers

### CODE REVIEW

File: tools/apigee-sackmesser/cmd/export/export.sh [hunk 1]
```diff
@@ -23,7 +23,11 @@ source "$SCRIPT_FOLDER/../../lib/logutils.sh"
 
 mgmtAPIDownload() {
     loginfo "Sackmesser export (zip) $1"
-    curl -fsS -H "Authorization: Bearer $token" "https://$baseuri/v1/$1" -o "$2"
+    if [ "$opdk" == "yes" ]; then
+        curl -fsS -u $username:$password -v "http://$baseuri/v1/$1" -o "$2"
```
Comment (g-greatdevaks) [inline tools/apigee-sackmesser/cmd/export/export.sh:27]: @SamPriyadarshi Let's stick to using Basic Auth here.

File: tools/apigee-sackmesser/bin/sackmesser [hunk 2]
```diff
@@ -158,6 +165,7 @@ if [[ -z "$token" && "$apiversion" = "apigee" ]]; then
       --data "grant_type=password")
     token=$(echo "$response" | jq -r '.access_token')
     export token
+    echo "token from sackmesser: $token"
```
Comment (g-greatdevaks) [inline tools/apigee-sackmesser/bin/sackmesser:168]: @SamPriyadarshi Let's remove printing of the Token.

File: tools/apigee-sackmesser/bin/sackmesser [hunk 3]
```diff
@@ -141,12 +143,17 @@ else
...
+
 if [[ -z "$token" && (-z "$username"  || -z "$password") ]]; then
     logfatal "required either -t (OAuth2 or GCP access token) or -u and -p (Edge username and password)"
     exit 1
 fi
 
-if [[ -z "$token" && "$apiversion" = "apigee" ]]; then
+if [[ -z "$token" && "$apiversion" = "apigee" && "$opdk" = "yes" ]]; then
+    token=`echo -n $username:$password | base64`
+    export $token
```
Comment (g-greatdevaks) [inline tools/apigee-sackmesser/bin/sackmesser:155]: Let's add log statement here.

File: tools/apigee-sackmesser/cmd/list/list.sh [hunk 4]
```diff
@@ -50,4 +50,9 @@ elif [ "$apiversion" = "apigee" ]; then
     esac
 fi
 
-curl -fsS -H "Authorization: Bearer $token" "https://$baseuri/v1/$partial_uri" | jq "$jq_pattern"
+if [ "$opdk" == "yes" ]; then
+    token=`echo -n $username:$password | base64`
+    curl -fsS -H "Authorization: Basic $token" -v "http://$baseuri/v1/$partial_uri" | jq "$jq_pattern" #to do: logic for https
```
Comment (g-greatdevaks) [inline tools/apigee-sackmesser/cmd/list/list.sh:55]: @SamPriyadarshi Let's add support for http/https both by using an if condition based on some flag like `--insecure`.

File: tools/apigee-sackmesser/bin/sackmesser (hunk 2 above)
Comment (SamPriyadarshi) [reply tools/apigee-sackmesser/bin/sackmesser]: Remove this echo statement.

File: tools/apigee-sackmesser/cmd/export/export.sh (hunk 1 above)
Comment (SamPriyadarshi) [reply tools/apigee-sackmesser/cmd/export/export.sh]: Added Basic auth.

File: tools/apigee-sackmesser/bin/sackmesser (hunk 3 above)
Comment (SamPriyadarshi) [reply tools/apigee-sackmesser/bin/sackmesser:156]: Added logdebug here.

**Review by danistrebel — APPROVED** [review]

**Pull request merged**